dotnet/src/SemanticKernel.Skills/Skills.Web/WebSearchEngineSkill.cs

```diff
@@ -20,14 +40,35 @@ public WebSearchEngineSkill(IWebSearchEngineConnector connector)
 
     [SKFunction("Perform a web search.")]
     [SKFunctionInput(Description = "Text to search for")]
+    [SKFunctionContextParameter(Name = Parameters.Count, Description = "Number of results")]
+    [SKFunctionContextParameter(Name = Parameters.Offset, Description = "Number of results to skip")]
     public async Task<string> SearchAsync(string query, SKContext context)
     {
-        string result = await this._connector.SearchAsync(query, context.CancellationToken);
-        if (string.IsNullOrWhiteSpace(result))
+        var count = context.Variables.ContainsKey(Parameters.Count) ? context[Parameters.Count] : "1";        
+        if (string.IsNullOrWhiteSpace(count)) { count = DefaultCount; }
+
+        var offset = context.Variables.ContainsKey(Parameters.Offset) ? context[Parameters.Offset] : "1";
+        if (string.IsNullOrWhiteSpace(offset)) { offset = DefaultOffset; }
+
+        int countInt = int.Parse(count, CultureInfo.InvariantCulture);
+        int offsetInt = int.Parse(offset, CultureInfo.InvariantCulture);
+        var  results = await this._connector.SearchAsync(query, countInt, offsetInt, context.CancellationToken);
+        if (!results.Any())
         {
             context.Fail("Failed to get a response from the web search engine.");
         }
 
-        return result;
+        string resultString;
+        if (countInt == 1)
+        {
+            var result = results.FirstOrDefault();
```

Comment:
these lines can be simplified to `resultString = results.FirstOrDefault() ?? string.Empty;`